backend/app/rabbitmq/message_listener_sync.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import pika
+import json
+import asyncio
+import random
+import string
+import datetime
+from packaging import version
+from pymongo import MongoClient
+from bson import ObjectId
+
+from app.config import settings
+from app.models.config import ConfigEntryDB, ConfigEntryOut
+from app.models.listeners import EventListenerDB, EventListenerOut, ExtractorInfo, \
+    EventListenerJob, EventListenerJobUpdate
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def callback(ch, method, properties, body):
+    """This method receives messages from RabbitMQ and processes them."""
+    msg = json.loads(body.decode("utf-8"))
+
+    job_id = msg["job_id"]
+    status = msg["status"]
+    timestamp = datetime.strptime(msg["start"], '%Y-%m-%dT%H:%M:%S%z')  # incoming format: '2023-01-20T08:30:27-05:00'
+
+    mongo_client = MongoClient(settings.MONGODB_URL)
+    db = mongo_client[settings.MONGO_DATABASE]
+
+    # Check if the job exists, and update if so
+    existing_job = db["listener_jobs"].find_one({"_id": ObjectId(job_id)})
```

Comment:
maybe instead of update it just keep storing them? This way we can have a complete history of the job status?

Reply:
the individual events are also stored in db["listener_job_updates"] but the "listener_jobs" just contain the "header" of each job, with the latest message/status, so we have both if I understand correctly.
```
# Add latest message to the job updates
        event_msg = EventListenerJobUpdate(job_id=job_id, status=status, timestamp=timestamp)
        db["listener_job_updates"].insert_one(event_msg.to_mongo())```